### ISSUE TITLE
Review contatenation logic in the Wasm SDK

### ISSUE THREAD

**Issue by javiercn** [dotnet/runtime#103783]
There are several more similar concatenations with `$(PublishDir)` and `$(OutDir)`. Do we need to address those as well?

_Originally posted by @maraf in https://github.com/dotnet/runtime/issues/103768#issuecomment-2181342715_

**Comment by maraf** [comment]
Moving to 10 as the remaining places don't block anything now

**Comment by pavelsavara** [comment]
is this dupe ? https://github.com/dotnet/runtime/issues/99292

**Comment by maraf** [comment]
> is this dupe ? [#99292](https://github.com/dotnet/runtime/issues/99292)

No. This one is about how we concatenate paths in MSBuild. The other one is about what is printed to the console after successful publish
